plugins/Feedback/API.php

```diff
@@ -57,7 +67,21 @@ public function sendFeedbackForFeature($featureName, $like, $message = false)
             empty($feedbackMessage) ? "" : "(w/ feedback)"
         );
 
-        $this->sendMail($subject, $body);
+        // Determine where Matomo is running and add as source
+        if (Config::getHostname() == 'demo.matomo.cloud') {
+            $source = 'Demo';
+        } else if (SettingsServer::isMatomoForWordPress()) {
+            $source = 'Wordpress';
+        } else {
+            $source = 'On-Premise';
+        }
+        $body .= "Source: ".$source."\n";
+
+        if ($message != 'test') {
```

Comment:
Was this for testing purpose, or do we try to avoid too many people sending messages while testing the feature?

Reply:
This is so the feature can be tested without spamming the feedback mailbox.

Reply:
For local testing purpose I would suggest to configure your mail settings in a way that prevent sending out any emails. Personally I'm using a mailcatcher to be able to see all outgoing mails instead of sending them

Reply:
I also use mailcatcher, but this isn't for manual tests. The UI tests also use the 'test' value and since the UI tests are publicly available we've no way of knowing who will run them and if their environment is setup to block mail sending, so this provides another way of blocking unwanted spamming of the feedback mailbox.

Reply:
Ok I see. Actually tests shouldn't sent out any mails. They should instead be written to a tmp file. See
https://github.com/matomo-org/matomo/blob/185e41fe7717d0018cec7deb073d0cfc5de8334f/core/Mail/Transport.php#L95-L102
https://github.com/matomo-org/matomo/blob/ddfa42e16f6648a1f54d2912fcc897e0ae07b75e/config/environment/test.php#L123-L135

Reply:
Nice, looks like I'm trying to solve problems that don't exist. :slightly_smiling_face:  I've removed this check.